src/state_machine/coordinator/fire.rs

```diff
@@ -182,11 +185,14 @@ impl<Aggregator: AggregatorTrait> Coordinator<Aggregator> {
                                 self.malicious_signer_ids.insert(*signer_id);
                             }
 
-                            let num_malicious_keys: u32 = self
+                            let num_malicious_keys: usize = self
                                 .malicious_signer_ids
                                 .iter()
-                                .map(|signer_id| self.config.signer_key_ids[signer_id].len() as u32)
+                                .map(|signer_id| {
+                                    self.config.public_keys.signer_key_ids[signer_id].len()
+                                })
                                 .sum();
```

Comment:
I don't think that we should use direct array access here. Instead we can do:
```suggestion
                                .flat_map(|signer_id| {
                                    self.config.public_keys.signer_key_ids.get(signer_id)
                                })
                                .map(|keys| keys.len())
                                .sum();
```
or maybe
```suggestion
                                .flat_map(|signer_id| {
                                    self.config.public_keys.signer_key_ids.get(signer_id)
                                })
                                .flatten()
                                .count();
```

Reply:
Interesting, I had no idea that `flat_map` would ignore errors.

Reply:
Fixed in e4b0ab401888e9db18d155c7c06af014fbfaf780

Reply:
Yeah, it filters out `Result::Err` and `Option::None`.